target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+            //TODO: expected program owner of the posted_vaa account should come from config account that can only be modified by governance
+            &wormhole_anchor_sdk::wormhole::program::id(),
+        );
+        require_keys_eq!(
+            vaa.key(),
+            expected_vaa_pubkey,
+            ReceiverError::InvalidVaaAccountKey
+        );
+
+        // TODO: expected emitter_chain should come from config account that can only be modified by governance
+        require_eq!(
+            vaa.emitter_chain(),
+            <wormhole::Chain as Into<u16>>::into(wormhole::Chain::Pythnet),
+            ReceiverError::InvalidEmitterChain
+        );
+
+        require_keys_eq!(
+            Pubkey::new_from_array(*vaa.emitter_address()),
+            // TODO: expected emitter_address should come from config account that can only be modified by governance
+            Pubkey::new_from_array(ACCUMULATOR_EMITTER_ADDRESS),
+            ReceiverError::InvalidEmitterAddress
+        );
+
+        let wh_message = WormholeMessage::try_from_bytes(vaa.payload.as_slice())
+            .map_err(|_| ReceiverError::InvalidWormholeMessage)?;
+        msg!("constructed wh_message {:?}", wh_message);
+        let root: MerkleRoot<Keccak160> = MerkleRoot::new(match wh_message.payload {
+            WormholePayload::Merkle(merkle_root) => merkle_root.root,
+        });
+
+        let mut count_updates = 0;
+
+        let price_updates_len = price_updates.len();
+        for price_update in price_updates {
+            let merkle_price_update =
+                from_slice::<byteorder::BE, MerklePriceUpdate>(price_update.as_slice())
+                    .map_err(|_| ReceiverError::DeserializeUpdateFailed)?;
+            let message_vec = Vec::from(merkle_price_update.message);
+            if !root.check(merkle_price_update.proof, &message_vec) {
+                return err!(ReceiverError::InvalidPriceUpdate);
+            }
+            let msg = from_slice::<byteorder::BE, Message>(&message_vec)
+                .map_err(|_| ReceiverError::DeserializeMessageFailed)?;
+
+            match msg {
+                Message::PriceFeedMessage(price_feed_message) => {
+                    count_updates += 1;
+                    msg!("price_feed_message: {:?}", price_feed_message);
+                }
+                Message::TwapMessage(twap_message) => {
+                    count_updates += 1;
+                    msg!("twap_message: {:?}", twap_message);
+                }
+                _ => return err!(ReceiverError::InvalidAccumulatorMessageType),
+            }
+        }
+        msg!("verified {} / {} updates", count_updates, price_updates_len);
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+#[instruction(vaa_hash: [u8; 32])]
+pub struct PostUpdates<'info> {
+    #[account(mut)]
+    pub payer:         Signer<'info>,
+    pub posted_vaa:    Box<Account<'info, AnchorVaa>>,
+    pub signature_set: Box<Account<'info, SignatureSetData>>,
```

Comment:
@jayantk @guibescos since this is using the `wormhole-anchor-sdk:SignatureSetData` account type, the owner is set to be the wormhole pid. since the `posted_vaa` pid/owner will eventually be checked using a config account, do we also want the signature set account owner to be configurable?